doc/examples/kinematics_cost_function/src/kinematics_cost_function_tutorial.cpp

```diff
@@ -0,0 +1,256 @@
+  copy_state->setFromIK(joint_model_group, target_pose, 0.0 /* timeout */, callback_fn, opts);
+
+  std::vector<double> solution;
+  current_state->copyJointGroupPositions(joint_model_group, solution);
+
+  std::vector<double> solution_no_cost_fn;
+  copy_state->copyJointGroupPositions(joint_model_group, solution_no_cost_fn);
+
+  // Now we can use our helper function from earlier to evaluate the effectiveness of the cost function.
+  double l2_solution = compute_l2_norm(solution, start_joint_positions);
+  RCLCPP_INFO_STREAM(LOGGER, "L2 norm of the solution WITH a cost function is " << l2_solution);
+  l2_solution = compute_l2_norm(solution_no_cost_fn, start_joint_positions);
+  RCLCPP_INFO_STREAM(LOGGER, "L2 norm of the solution WITHOUT a cost function is " << l2_solution);
+
+  // If we're happy with the solution, we can set it as a joint value target.
+  move_group.setJointValueTarget(solution);
+
+  // We lower the allowed maximum velocity and acceleration to 5% of their maximum.
+  // The default values are 10% (0.1).
+  // Set your preferred defaults in the joint_limits.yaml file of your robot's moveit_config
+  // or set explicit factors in your code if you need your robot to move faster.
+  move_group.setMaxVelocityScalingFactor(0.05);
+  move_group.setMaxAccelerationScalingFactor(0.05);
+
+  moveit::planning_interface::MoveGroupInterface::Plan my_plan;
+  bool success = (move_group.plan(my_plan) == moveit::core::MoveItErrorCode::SUCCESS);
+  RCLCPP_INFO(LOGGER, "Visualizing plan 1 (joint space goal with cost function) %s", success ? "" : "FAILED");
+
+  // Visualize the plan in RViz:
+  visual_tools.deleteAllMarkers();
+  visual_tools.publishAxisLabeled(target_pose, "goal_pose");
+  visual_tools.publishText(text_pose, "Joint_Space_Goal", rvt::WHITE, rvt::XLARGE);
+  visual_tools.publishTrajectoryLine(my_plan.trajectory, joint_model_group);
+  visual_tools.trigger();
+
+  // Uncomment if you would like to execute the motion
+  /*
+  visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to execute the motion");
+  move_group.execute(my_plan);
+  */
+
+  visual_tools.prompt("Press 'next' in the RvizVisualToolsGui window to continue.");
+
+  // We can also specify cost functions when computing robot trajectories that must follow a cartesian path.
+  // First, let's change the target pose for the end of the path, so that if the previous motion plan was executed,
+  // we still have somewhere to move.
+  target_pose.position.y += 0.4;
+  target_pose.position.z -= 0.1;
+  target_pose.orientation.w = 0;
+  target_pose.orientation.x = -1.0;
+  Eigen::Isometry3d target;
+  tf2::fromMsg(target_pose, target);
+
+  auto start_state = move_group.getCurrentState(10.0);
+  std::vector<moveit::core::RobotStatePtr> traj;
+  moveit::core::MaxEEFStep max_eef_step(0.01, 0.1);
+  // Here, were effectively disabling the jump threshold for joints. This is not recommended on real hardware.
```

Comment:
```suggestion
  // Here, we're effectively disabling the jump threshold for joints. This is not recommended on real hardware.
```

... but also, the jump threshold for what?

Reply:
The documentation for this is really good :ok_hand: 
```cpp
/** \brief Struct for containing jump_threshold.

    For the purposes of maintaining API, we support both \e jump_threshold_factor which provides a scaling factor for
    detecting joint space jumps and \e revolute_jump_threshold and \e prismatic_jump_threshold which provide absolute
    thresholds for detecting joint space jumps. */
```

```cpp
/**
     During the computation of the trajectory, it is usually preferred if consecutive joint values do not 'jump' by a
     large amount in joint space, even if the Cartesian distance between the corresponding points is small as expected.
     To account for this, the \e jump_threshold struct is provided, which comprises three fields:
     \e jump_threshold_factor, \e revolute_jump_threshold and \e prismatic_jump_threshold.
     If either \e revolute_jump_threshold or \e prismatic_jump_threshold  are non-zero, we test for absolute jumps.
     If \e jump_threshold_factor is non-zero, we test for relative jumps. Otherwise (all params are zero), jump
     detection is disabled.

     For relative jump detection, the average joint-space distance between consecutive points in the trajectory is
     computed. If any individual joint-space motion delta is larger then this average distance by a factor of
     \e jump_threshold_factor, this step is considered a failure and the returned path is truncated up to just
     before the jump.

     For absolute jump thresholds, if any individual joint-space motion delta is larger then \e revolute_jump_threshold
     for revolute joints or \e prismatic_jump_threshold for prismatic joints then this step is considered a failure and
     the returned path is truncated up to just before the jump.
     **/
```

Reply:
I understand that users can look at documentation for both these functions I commented on. I'm not asking for you to provide me with the answer to the question, but rather to help newcomers a little bit in the comments so they don't need to go into other files to get basic understanding of the tutorials. For example:

"we're effectively disabling the threshold for joint angle jumps during motion planning"

or

"if you want to compare results without the custom cost function, omit the last argument"

Anyways, these aren't blocking issues by any means. As you said, people can eventually figure this out by reading the source code.